nixos/modules/hardware/decklink.nix

```diff
@@ -0,0 +1,24 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.decklink;
+  kernelPackages = config.boot.kernelPackages;
+in
+{
+  options.hardware.decklink.enable = mkEnableOption "Enable hardware support for the Blackmagic Design Decklink audio/video interfaces.";
+
+  config = mkIf cfg.enable {
+    boot.kernelModules = [ "blackmagic" "blackmagic-io" "snd_blackmagic-io" ];
+    boot.extraModulePackages = [ kernelPackages.decklink ];
+
+    # supporting service
+    systemd.services."DecklinkVideoHelper" = {
```

Comment:
can this run as non-root?

Reply:
I'll have to check when testing on hardware.

Reply:
the helper isn't needed at all.

Reply:
Correction, it actually is. I had it running from a previous invocation. It also has to run as root.

Reply:
Really? What capabilities does it need? What files/devices does it access? I don't think a NixOS module should run some random proprietary binary as root service unless it absolutely has to.

Reply:
I can run an strace on it next time I have access to the hardware. It is unclear what it actually does, but without it running the device does not initialize. I assume it sends a bitstream to onboard FPGAs or something like that.

Reply:
That would be great!

Reply:
I've finally taken that strace, see https://gist.github.com/hexchen/cc39485001ba513e0cd9c970fd695f18

I am unsure on how to act on this though, as whatever it is doing only succeeds if run as root, and I am unable to figure out why from that strace.